go/private/rules/cgo.bzl

```diff
@@ -57,6 +57,40 @@ load(
 
 _CgoCodegen = provider()
 _CgoInfo = provider()
+_CcAlwaysLinkInfo = provider()
+
+_WHOLE_ARCHIVE_FLAGS_BY_GOOS = {
+    goos: ("-Wl,--whole-archive", "-Wl,--no-whole-archive")
+    for goos in GOOS.keys()
+} + {
+    "darwin": ("-Wl,-all_load", ""),
+    "js": ("", ""),
+    "windows": ("", ""),
```

Comment:
Out of curiosity, why aren't the flags needed on Windows? Might be good to have a comment explaining it.

Reply:
They are needed, but I'm not sure what they are

Reply:
I meant depending on MSVC or MSYS

Reply:
cgo doesn't support MSVC, so don't worry about that. I think MSYS should accept gcc flags, but I don't know if there's a slightly different variant on Windows. Maybe just take out the case for Windows, and I can fix it later if it's a problem.